samples: nrf9160: gnss: rename sample and add nRF Cloud A-GPS/P-GPS support (#6010)
nrfconnect/sdk-nrf: samples/nrf9160/gnss/README.rst

.. _gnss_sample:

nRF9160: GNSS
#############

.. contents::
   :local:
   :depth: 2

> The GNSS sample demonstrates how to retrieve `GNSS`_ data.

Reviewer: ```suggestion
This sample demonstrates how to retrieve `GNSS`_ data.
```

> It also shows how to improve fix speed and accuracy with :ref:`lib_nrf_cloud_agps` and how to use :ref:`lib_nrf_cloud_pgps`.

Reviewer: ```suggestion
It also shows how to improve fix speed and accuracy with the :ref:`lib_nrf_cloud_agps` library and how to use the :ref:`lib_nrf_cloud_pgps` library.
```

Assistance data is downloaded from nRF Cloud using `nRF Cloud's REST-based device API <nRF Cloud REST API_>`_.

> The sample also supports using SUPL client library for downloading A-GPS data from a SUPL server.

Reviewer: ```suggestion
The sample also supports using the SUPL client library for downloading A-GPS data from a SUPL server.
```

The sample can also be configured to run in NMEA-only mode, where only the NMEA strings are displayed in the console.
The NMEA-only mode can be used to visualize the data from the GNSS using a third-party tool.

> A-GPS and P-GPS support can be enabled for both the default mode (PVT and NMEA) and the NMEA-only mode.

Reviewer: ```suggestion
You can enable A-GPS and P-GPS support for both the default mode (PVT and NMEA) and the NMEA-only mode.
```

                $GPGSV,1,1,2,17,,,24,1,,,28,1*6D
                $GPRMC,000002.00,V,,,,,,,060180,,,N,V*0A

   If A-GPS and/or P-GPS support is enabled:

   a. Observe that the following message is displayed in the terminal emulator immediately after the device boots:

           .. code-block:: console

              Assistance data needed, ephe 0xffffffff, alm 0xffffffff, flags 0x3b

   b. Observe the following actions in the terminal emulator:

>       i. The sample downloads the requested assistance data if needed (with P-GPS the data may already be available in flash memory).

Reviewer: ```suggestion
      i. The sample downloads the requested assistance data if needed (with P-GPS, the data may already be available in the flash memory).
```

>       #. The sample continues after the download has been completed.

Reviewer: ```suggestion
      #. The sample continues after the download has completed.
```
